src/components/WeeklySchedule.js

```diff
@@ -0,0 +1,97 @@
+import PropTypes from "prop-types";
+import WeeklyShow from "./WeeklyShow";
+import { getDayInt } from "../lib/globals.js";
+import styles from "../styles/WeeklyShow.module.css";
+
+export default function WeeklySchedule({ shows }){
+  const showsArr = [];
+  const days = ["","Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"];
+
+  const firstRow = days.map((d, i) => {
+    const currKey = `firstrow${  i}`;
+    return (<div key={currKey} className={styles.scheduleDay}>
+              {d}
+            </div>)
+  })
+
+  showsArr.push(firstRow);
+
+  // Add useEffect when writing the onHover function
+  for (let i = 0; i < 24; i++){
+    showsArr.push([]);
+    for (let l = 0; l < 8; l++){
+      if (l===0){
+        showsArr[showsArr.length-1].push(<div className={styles.scheduleTime}>{`${i}:00`}</div>);
+      }
+      else {showsArr[showsArr.length-1].push(undefined)}
+    }
+  }
+  
+  shows.forEach((s) => {
+    const day = getDayInt(s.time.day)+1;
+    const time = (s.time.hour / 100)+1;
+    showsArr[time][day] = <WeeklyShow show = {s}/>
+    if (s.time.duration === 2) {
+      showsArr[time+1][day] = <WeeklyShow show = {s}/>
+    }
+  })
+
+
+  const table = showsArr.map((item, key1) => {
```

Comment:
It took me a really long time to figure out what exactly is happening in this function. I think I've figured it out, but I can't help but wonder if there is an easier way to construct the table.

It looks like you are jumping through some hoops to give all the elements keys, which is where I got confused. It might be easiest to just use the show id as a key for each cell.

Reply:
I guess the problem with using show ids for each cell is some (nearly half) don't have shows in them. 
I also agree there must be an easier way to do this. The only thing I can think of that would be 'cleaner' would also involve a ton of looping and mapping. I wonder if this could change once we have a server and this whole loading thing doesn't have to happen on the users browser.

Reply:
Good point about the empty cells. Perhaps we could revisit this in the future